sphinxcontrib/aafig.py

```diff
@@ -86,7 +86,7 @@ def run(self):
                 if k == 'scale' or k == 'aspect':
                     v = float(v) / 100.0
                 aafig_options[k] = v
-                del self.options[k]
+                #del self.options[k]
```

Comment:
Why is this commented out?

Reply:
Hey, sorry I totally overlooked this issue (it was a long day for me, but I wanted to contribute the fixes that worked for me). My Python interpreter threw an exception that the dictionary is altered during iteration; I couldn't find a reason why the option should be deleted after processing them, so I commented-out that line and I was able to use aafig :)

Reply:
Ah, I see. I think the options are removed because they are *consumed* by this extension, so they are not passed to other extensions that might not know them. That said I haven't used (or developed for) Sphinx for probably more than a decade now, so my knowledge about the internals are very limited.

One easy way to fix the *dictionary is altered during iteration* issue is to iterate over a copy of the keys:

```py
for k in tuple(self.options.keys()):
    v = self.options[k]
    ...  # Rest of the for loop
```
This way we keep the same logic (that seems to be working so far).